### ISSUE TITLE
Support for xgboost, lightgbm and catboost

### ISSUE THREAD

**Issue by shankarpandala** [shankarpandala/lazypredict#54]
Support for xgboost, lightgbm and catboost

**Comment by shankarpandala** [comment]
Completed

**Comment by jmakov** [comment]
Running the example I don't see catboost in the table. Am I missing something?
```
Model                                                                           
LinearSVC                          0.99               0.99     0.99      0.99   
Perceptron                         0.99               0.98     0.98      0.99   
LogisticRegression                 0.99               0.98     0.98      0.99   
SVC                                0.98               0.98     0.98      0.98   
XGBClassifier                      0.98               0.98     0.98      0.98   
LabelPropagation                   0.98               0.97     0.97      0.98   
LabelSpreading                     0.98               0.97     0.97      0.98   
BaggingClassifier                  0.97               0.97     0.97      0.97   
PassiveAggressiveClassifier        0.98               0.97     0.97      0.98   
SGDClassifier                      0.98               0.97     0.97      0.98   
RandomForestClassifier             0.97               0.97     0.97      0.97   
CalibratedClassifierCV             0.98               0.97     0.97      0.98   
LGBMClassifier                     0.97               0.97     0.97      0.97   
QuadraticDiscriminantAnalysis      0.96               0.97     0.97      0.97   
ExtraTreesClassifier               0.97               0.96     0.96      0.97   
RidgeClassifierCV                  0.97               0.96     0.96      0.97   
RidgeClassifier                    0.97               0.96     0.96      0.97   
AdaBoostClassifier                 0.96               0.96     0.96      0.96   
KNeighborsClassifier               0.96               0.96     0.96      0.96   
BernoulliNB                        0.95               0.95     0.95      0.95   
LinearDiscriminantAnalysis         0.96               0.95     0.95      0.96   
GaussianNB                         0.95               0.95     0.95      0.95   
NuSVC                              0.95               0.94     0.94      0.95   
ExtraTreeClassifier                0.94               0.93     0.93      0.94   
NearestCentroid                    0.95               0.93     0.93      0.95   
DecisionTreeClassifier             0.93               0.93     0.93      0.93   
DummyClassifier                    0.64               0.50     0.50      0.50 
```
example:
```python
from lazypredict.Supervised import LazyClassifier
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split

data = load_breast_cancer()
X = data.data
y= data.target

X_train, X_test, y_train, y_test = train_test_split(X, y,test_size=.5,random_state =123)

clf = LazyClassifier(verbose=0,ignore_warnings=True, custom_metric=None)
models,predictions = clf.fit(X_train, X_test, y_train, y_test)

print(models)
```

**Comment by shankarpandala** [comment]
@jmakov catboost is optional requirement
if you have catboost already installed in the environment lazypredict with use it else it will ignore.
I suggest not to use catboost with lazypredict

**Comment by jmakov** [comment]
@shankarpandala why not?

**Comment by BrandonKMLee** [comment]
Seconded, although CatBoost has more parameters, it also has higher accuracy than XGBoost & faster than Gradient Boosting proper.
